flash/image/classification/integrations/labelstudio/handler.py

```diff
@@ -0,0 +1,138 @@
+
+    _IMAGE_CACHE_DIR: str
+
+    def __init__(self, source: str, **kwargs):
+        super().__init__(**kwargs)
+        self.from_name, self.to_name, self.value, self.classes = get_single_tag_keys(
+            self.parsed_label_config, "Choices", "Image"
+        )
+
+        num_classes = len(self.classes)
+
+        head = torch.nn.Sequential(
+            torch.nn.Dropout(p=0.1),
+            torch.nn.Linear(512, num_classes),
+        )
+
+        self.model = ImageClassifier(
+            backbone="resnet18", num_classes=num_classes, head=head, serializer=Probabilities()
+        )
+
+        self.heuristic = BALD()
+
+        self._IMAGE_CACHE_DIR = os.path.join(os.path.dirname(source), "image_cache")
+        os.makedirs(self._IMAGE_CACHE_DIR, exist_ok=True)
+
+    def predict(self, tasks, **kwargs):
+        image_urls = [self._firstv(task["data"]) for task in tasks]
+        predict_files = [self._download_image(url) for url in image_urls]
+
+        datamodule = ImageClassificationData.from_files(predict_files=predict_files)
+        trainer = flash.Trainer(max_epochs=1, gpus=int(torch.cuda.is_available()))
+        predictions = trainer.predict(self.model, datamodule=datamodule)
+        predictions = torch.tensor(predictions)
+        predicted_label_indices = predictions[0].argmax(-1)
+        predicted_scores = self.heuristic.get_uncertainties(predictions)
+
+        predictions = []
+        for idx, score in zip(predicted_label_indices, predicted_scores):
+            predicted_label = self.classes[idx]
+            # prediction result for the single task
+            result = [
+                {
+                    "from_name": self.from_name,
+                    "to_name": self.to_name,
+                    "type": "choices",
+                    "value": {"choices": [predicted_label]},
+                }
+            ]
+
+            # expand predictions with their scores for all tasks
+            predictions.append({"result": result, "score": float(score)})
+
+        return predictions
+
+    def fit(self, completions, workdir=None, batch_size=32, num_epochs=10, **kwargs):
+        image_urls, image_classes = [], []
+        for completion in completions:
+
+            if is_skipped(completion):
+                continue
+            image_urls.append(completion["data"][self.value])
+            image_classes.append(get_choice(completion))
+
+        train_files = [self._download_image(url, url_class) for url, url_class in zip(image_urls, image_classes)]
+        class_to_id = {c: id for id, c in enumerate(np.unique(image_classes))}
+        train_targets = np.vectorize(class_to_id.get)(image_classes)
+        datamodule = ImageClassificationData.from_files(train_files=train_files, train_targets=train_targets)
+        trainer = flash.Trainer(max_epochs=1, gpus=int(torch.cuda.is_available()))
+        trainer.fit(self.model, datamodule=datamodule)
+        model_path = os.path.join(".", "model.pt")
```

Comment:
there is a `workdir` folder that serves for storing checkpoints from the current run. Unless you want to overwrite the same model.pt file each time, I would recommend to use `os.path.join(workdir, "model.pt")`

Reply:
Done !